lib/code-ribbon-ribbon-container.js

```diff
@@ -135,28 +152,89 @@ class CodeRibbonRibbonContainer extends PaneAxis {
       n_crsr.initialize();
     }
 
+    // increment / remove patchQuota
+    if (
+      this.autoincrement_patch_quota() == false
+    ) {
+      this.autoremove_excess_empty_right_patch_column();
+    }
+
     // update all the children
     this.children.map(child => {
       child.cr_update();
     });
 
     // set width and scrolling etc,
-    // find the max number of items of any child:
-    this.max_horiz_patches = 1;
-    // crdebug("Finding max_horiz_patches...")
+    var widthpercent = 100.0 * (
+      this.patchQuota / atom.config.get("code-ribbon.pane_count_calc.pane_count_horizontal_number")
+    );
+    this.getElement().style.width = widthpercent.toString() + "%";
+
+    // TODO deal with case when we have too many Ribbons for target count
+
+  }
+
+  cr_add_column_right() {
+    // patches add empty panes right to meet quota
+    this.patchQuota++;
+    this.cr_update();
+  }
+
+  /**
+   * Increases the patch quota so that there's at one screen of patches -1
+   * of patches being renderend after the last contentful patch
+   * @return {bool} true if patchQuota was increased
+   */
+  autoincrement_patch_quota() {
+    // check if we need to increase patchQuota
+    // find the max number of real patches of any child:
+    var child_ribbon_max_patch_count = 1;
+    // crdebug("Finding child_ribbon_max_patch_count...")
     this.children.map(child => {
       // crdebug("Looking at ", child);
-      if (child.children.length > this.max_horiz_patches) {
-        this.max_horiz_patches = child.children.length;
+      if (child.get_contentful_length() > child_ribbon_max_patch_count) {
+        child_ribbon_max_patch_count = child.get_contentful_length();
```

Comment:
what is contentful length?

Reply:
Contentful length is a metric used to avoid counting empty patches at the end of a ribbon. (It still counts empty patches before contentful ones...)